unzip/bsdunzip.c

```diff
@@ -884,22 +885,25 @@ list(struct archive *a, struct archive_entry *e)
 	else
 		strftime(buf, sizeof(buf), "%m-%d-%g %R", tm);
 
+	pathname = archive_entry_pathname(e);
+	if (!pathname)
+		pathname = "INVALID PATH";
```

Comment:
Your approach here is reasonable, but I would use a different string here.  Perhaps "-" or "<empty>" to indicate the lack of a pathname.  Zip and other file formats do technically allow for an empty pathname, so the word "INVALID" is a bit misleading.